### PR TITLE
Adding microtime to the constructor

### PR DESCRIPTION
**Pull request by bassrock**
I have added microtome to the Carbon date time constructor if the time parameter is null or now.

### CODE REVIEW

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -180,7 +180,7 @@ protected static function safeCreateDateTimeZone($object)
      */
     public function __construct($time = null, $tz = null)
     {
-        // If the class has a test now set and we are trying to create a now()
+                // If the class has a test now set and we are trying to create a now()
```
Comment (GrahamCampbell) [inline src/Carbon/Carbon.php]: ?

File: src/Carbon/Carbon.php [hunk 2]
```diff
@@ -198,7 +198,16 @@ public function __construct($time = null, $tz = null)
             $time = $testInstance->toDateTimeString();
         }
 
-        parent::__construct($time, static::safeCreateDateTimeZone($tz));
+        if(is_null($time) || $time == 'now') {
```
Comment (GrahamCampbell) [inline src/Carbon/Carbon.php]: cs

File: src/Carbon/Carbon.php [hunk 3]
```diff
@@ -198,7 +198,16 @@ public function __construct($time = null, $tz = null)
             $time = $testInstance->toDateTimeString();
         }
 
-        parent::__construct($time, static::safeCreateDateTimeZone($tz));
+        if (is_null($time) || $time == 'now') {
```
Comment (dereuromark) [inline src/Carbon/Carbon.php]: Comparison should be strict ===

**Comment by bassrock** [comment]
@GrahamCampbell any chance on getting this merged?

**Pull request closed without merging**